rum-injection-installer/installer/nginx_configurator.go

```diff
@@ -0,0 +1,272 @@
+	newContent.WriteString(fmt.Sprintf("load_module %s/ngx_http_datadog_module.so;\n", n.ModulesPath))
+
+	// Find http and bracket { on possibly multiple lines
+	httpRegex := regexp.MustCompile(`(?m)^[\s]*http[\s]*\{`)
+
+	httpBlockStart := httpRegex.FindIndex(content)
+	if httpBlockStart == nil {
+		return fmt.Errorf("could not find http block in nginx config")
+	}
+
+	newContent.Write(content[:httpBlockStart[0]])
+
+	newContent.Write(content[httpBlockStart[0]:httpBlockStart[1]])
+	newContent.WriteString("\n")
+
+	datadogConfig := fmt.Sprintf(`    datadog_agent_url %s;
+    datadog_disable;
+    datadog_rum on;
+    datadog_rum_config "v5" {
+        "applicationId" "%s";
+        "clientToken" "%s";
+        "site" "%s";
+        "sessionSampleRate" %d;
+        "sessionReplaySampleRate" %d;
+    }
+`, agentUrl, appID, clientToken, site, sessionSampleRate, sessionReplaySampleRate)
+	newContent.WriteString(datadogConfig)
+
+	newContent.Write(content[httpBlockStart[1]:])
+
+	// Write the modified content back to the file
+	if err := os.WriteFile(configPath, []byte(newContent.String()), 0644); err != nil {
+		return fmt.Errorf("failed to write modified config: %v", err)
+	}
+
+	fmt.Printf("Successfully modified NGINX configuration file: %s\n", configPath)
+	return nil
+}
+
+// Runs 'nginx -t' to validate the configuration
+func (n *NginxConfigurator) ValidateConfig() error {
+	cmd := exec.Command("nginx", "-t")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("nginx configuration validation failed with output:\n%s", output)
+	}
+	return nil
+}
+
+// Downloads the latest nginx-datadog for the specified architecture
+// Extracts the .so file and places it in the modules path
+func (n *NginxConfigurator) DownloadAndInstallModule(arch string) error {
+	if err := n.getModulesPath(); err != nil {
+		return err
+	}
+
+	moduleURL := fmt.Sprintf("https://github.com/DataDog/nginx-datadog/releases/latest/download/ngx_http_datadog_module-%s-%s.so.tgz", arch, n.Version)
+
+	moduleContent, err := downloadFile(moduleURL)
+	if err != nil {
+		return fmt.Errorf("failed to download module: %v", err)
+	}
+
+	signatureURL := moduleURL + ".asc"
+	signatureContent, err := downloadFile(signatureURL)
+	if err != nil {
+		return fmt.Errorf("failed to download signature: %v", err)
+	}
+
+	publicKeyURL := "https://github.com/DataDog/nginx-datadog/releases/latest/download/pubkey.gpg"
+	publicKeyContent, err := downloadFile(publicKeyURL)
+	if err != nil {
+		return fmt.Errorf("failed to download public key: %v", err)
+	}
+
+	tmpDir, err := os.MkdirTemp("", "nginx-module-verification")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	moduleFile := filepath.Join(tmpDir, "module.tgz")
+	signatureFile := filepath.Join(tmpDir, "module.tgz.asc")
+	publicKeyFile := filepath.Join(tmpDir, "pubkey.gpg")
+
+	if err := os.WriteFile(moduleFile, moduleContent, 0644); err != nil {
```

Comment:
**[question]** Mind explaining why `0644`? Is it required?

Reply:
It is required for this API, which writes to file in a single call. There's another API that would let us not specify permissions, instead uses a default 666 for all files created,but we need to do more error handling, defer closing the files... Another thing to mention is `WriteFile` doesn't change permissions if the file already exists, so we are not messing with their actual config file. 644 is just the safest permissions I could think about that wouldn't cause problems, but I'm happy with 666 as well if you'd rather have that